src/protocol/sort/shuffle.rs

```diff
@@ -110,7 +110,7 @@ async fn reshare_all_shares<F: Field>(
 #[allow(clippy::cast_possible_truncation)]
 async fn shuffle_or_unshuffle_once<F: Field>(
     mut input: Vec<Replicated<F>>,
-    random_permutations: (&[u32], &[u32]),
+    random_permutations: &(Vec<u32>, Vec<u32>),
```

Comment:
Why make this change?

Reply:
aesthetically this is easier to read than passing two pointers each time

Reply:
It is very different from a functional perspective though.  In order to create the latter argument, you have to move the vector into a tuple, then reference that tuple.  In the former, you only need to create a tuple from two references.

As always, you should prefer to pass slices of things when that is what is needed.  If you need flexibility, there is also `impl IntoIterator<Item = F>`.

Reply:
#265 Although I havent used the impl since the conversion needed is (Vec<..>, Vec<..>) -> (&[..], &[..]